### PR TITLE
DCOS-41313: telling app tasks from pods tasks

### PR DESCRIPTION
**Pull request by nLight**
Current strategy is not bulletproof. 
Container of a pod can have the same `task.name` as a standalone service thus making it ambiguous.
With this PR I attempted to use `task.id` instead of `task.name`.

## Testing

Services to launch:
```
{
  "id": "/nginx",
  "backoffFactor": 1.15,
  "backoffSeconds": 1,
  "container": {
    "portMappings": [],
    "type": "MESOS",
    "volumes": [],
    "docker": {
      "image": "nginx",
      "forcePullImage": false,
      "parameters": []
    }
  },
  "cpus": 0.1,
  "disk": 0,
  "instances": 2,
  "maxLaunchDelaySeconds": 300,
  "mem": 128,
  "gpus": 0,
  "networks": [
    {
      "name": "dcos",
      "mode": "container"
    }
  ],
  "requirePorts": false,
  "upgradeStrategy": {
    "maximumOverCapacity": 1,
    "minimumHealthCapacity": 1
  },
  "killSelection": "YOUNGEST_FIRST",
  "unreachableStrategy": {
    "inactiveAfterSeconds": 0,
    "expungeAfterSeconds": 0
  },
  "healthChecks": [],
  "fetch": [],
  "constraints": []
}
```

```
{
  "id": "/pod/nginx",
  "version": "2019-04-05T12:12:29.419Z",
  "containers": [
    {
      "name": "nginx",
      "resources": {
        "cpus": 0.001,
        "mem": 32,
        "disk": 0,
        "gpus": 0
      },
      "exec": {
        "command": {
          "shell": "sleep infinity"
        }
      }
    },
    {
      "name": "web",
      "resources": {
        "cpus": 0.001,
        "mem": 32,
        "disk": 0,
        "gpus": 0
      },
      "exec": {
        "command": {
          "shell": "sleep infinity"
        }
      }
    },
    {
      "name": "mysql",
      "resources": {
        "cpus": 0.001,
        "mem": 32,
        "disk": 0,
        "gpus": 0
      },
      "exec": {
        "command": {
          "shell": "sleep infinity"
        }
      }
    }
  ],
  "networks": [
    {
      "mode": "host"
    }
  ],
  "scaling": {
    "instances": 1,
    "kind": "fixed"
  },
  "scheduling": {
    "backoff": {
      "backoff": 1,
      "backoffFactor": 1.15,
      "maxLaunchDelay": 300
    },
    "upgrade": {
      "minimumHealthCapacity": 1,
      "maximumOverCapacity": 1
    },
    "killSelection": "YOUNGEST_FIRST",
    "unreachableStrategy": {
      "inactiveAfterSeconds": 0,
      "expungeAfterSeconds": 0
    },
    "placement": {
      "constraints": []
    }
  },
  "executorResources": {
    "cpus": 0.1,
    "mem": 32,
    "disk": 10
  },
  "volumes": [],
  "fetch": []
}
```

Verify that `nginx` service doesn't have `pod_nginx.instance-*` tasks
While `/pod/nginx` has it's containers

## Screenshots

Screenshot 2019-04-05 at 16 03 39

### CODE REVIEW

**Comment by GeorgiSTodorov** [comment]
Looks good. Some integration tests are failing, I will take a look to see how can we fix them.

**Comment by GeorgiSTodorov** [comment]
I fixed it, @nLight , do you mind if I push my fix?

**Comment by nLight** [comment]
@GeorgiSTodorov please go ahead 🙇

**Comment by GeorgiSTodorov** [comment]
If I broke any other tests while fixing `ServiceDetail-cy`, I will take responsibility for fixing them.

**Comment by GeorgiSTodorov** [comment]
I've fixed all failing tests, except one, currently working on this last one.

**Comment by GeorgiSTodorov** [comment]
I can't understand why the framework is not healthy in the test.
Screenshot from 2019-04-09 14-45-26

**Comment by nLight** [comment]
Thanks for picking up this one! Let me check the last failing test...

**Comment by GeorgiSTodorov** [comment]
No problem, I am also currently investigating. I will push my commit fixing the other tests.

**Comment by nLight** [comment]
@GeorgiSTodorov could that be because of the feature Pierre and Philipp are working on?

**Comment by GeorgiSTodorov** [comment]
@nLight , I think it's because of the fixtures, but it could be because of the feature.

File: tests/_fixtures/healthy-tasks-in-mesos-and-marathon/groups.json [hunk 1]
```diff
@@ -148,7 +130,7 @@
           ],
           "startedAt": "2017-01-09T20:02:26.435Z",
           "version": "2017-01-09T20:02:12.029Z",
-          "id": "confluent-kafka.825e1e2e-d6a6-11e6-a564-8605ecf0a9df",
+          "id": "confluent-kafka.instance-825e1e2e-d6a6-11e6-a564-8605ecf0a9df._app.1",
```
Comment (nLight) [inline tests/_fixtures/healthy-tasks-in-mesos-and-marathon/groups.json]: @GeorgiSTodorov looks like this one was the missing bit

Comment (GeorgiSTodorov) [reply tests/_fixtures/healthy-tasks-in-mesos-and-marathon/groups.json]: Interesting. Thanks for fixing it.

**Pull request merged**